### ISSUE TITLE
ConcurrentModificationError: Concurrent modification during iteration

### ISSUE THREAD

**Issue by pedia** [renefloor/flutter_cache_manager#6]
I got some sentry report with Exception maybe about flutter_cache_manager.

```
ConcurrentModificationError: Concurrent modification during iteration: _LinkedHashMap len:204.
  File "hooks.dart", line 109, in _drawFrame
  File "hooks.dart", line 122, in _invoke
  File "zone.dart", line 925, in _CustomZone.runGuarded
  File "zone.dart", line 1023, in _CustomZone.run
  File "zone.dart", line 1126, in _rootRun
  File "binding.dart", line 842, in BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame
  File "binding.dart", line 930, in BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame
  File "binding.dart", line 990, in BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback
  File "binding.dart", line 208, in BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding._handlePersistentFrameCallback
  File "binding.dart", line 622, in BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding&WidgetsBinding.drawFrame
  File "binding.dart", line 272, in BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding.drawFrame
  File "object.dart", line 796, in PipelineOwner.flushPaint
  File "object.dart", line 110, in PaintingContext.repaintCompositedChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "layout_builder.dart", line 222, in _RenderLayoutBuilder.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "shifted_box.dart", line 70, in RenderShiftedBox.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 128, in PaintingContext.paintChild
  File "object.dart", line 147, in PaintingContext._compositeChild
  File "object.dart", line 110, in PaintingContext.repaintCompositedChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "viewport.dart", line 330, in RenderViewportBase.paint
  File "object.dart", line 322, in PaintingContext.pushClipRect
  File "object.dart", line 305, in PaintingContext.pushLayer
  File "viewport.dart", line 339, in RenderViewportBase._paintContents
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "sliver_padding.dart", line 300, in RenderSliverPadding.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "sliver_multi_box_adaptor.dart", line 557, in RenderSliverMultiBoxAdaptor.paint
  File "object.dart", line 128, in PaintingContext.paintChild
  File "object.dart", line 147, in PaintingContext._compositeChild
  File "object.dart", line 110, in PaintingContext.repaintCompositedChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "shifted_box.dart", line 70, in RenderShiftedBox.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 1863, in RenderDecoratedBox.paint
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "shifted_box.dart", line 70, in RenderShiftedBox.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "flex.dart", line 931, in RenderFlex.paint
  File "box.dart", line 2196, in RenderBox&ContainerRenderObjectMixin&RenderBoxContainerDefaultsMixin.defaultPaint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 125, in RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.paint
  File "object.dart", line 130, in PaintingContext.paintChild
  File "object.dart", line 2018, in RenderObject._paintWithContext
  File "proxy_box.dart", line 1844, in RenderDecoratedBox.paint
  File "box_decoration.dart", line 415, in _BoxDecorationPainter.paint
  File "box_decoration.dart", line 397, in _BoxDecorationPainter._paintBackgroundImage
  File "decoration_image.dart", line 239, in DecorationImagePainter.paint
  File "image_provider.dart", line 264, in ImageProvider.resolve
  File "synchronous_future.dart", line 38, in SynchronousFuture.then
  File "image_provider.dart", line 266, in ImageProvider.resolve.<fn>
  File "image_cache.dart", line 82, in ImageCache.putIfAbsent
  File "image_provider.dart", line 266, in ImageProvider.resolve.<fn>.<fn>
  File "cached_network_image.dart", line 40, in CachedNetworkImageProvider.load
  File "<asynchronous suspension>"
  File "cached_network_image.dart", line 51, in CachedNetworkImageProvider._loadAsync
  File "<asynchronous suspension>"
  File "flutter_cache_manager.dart", line 227, in CacheManager.getFile
  File "<asynchronous suspension>"
  File "flutter_cache_manager.dart", line 77, in CacheManager._save
  File "<asynchronous suspension>"
  File "flutter_cache_manager.dart", line 136, in CacheManager._cleanCache
  File "synchronized.dart", line 33, in synchronized
  File "synchronized_impl.dart", line 225, in synchronized
  File "synchronized_impl.dart", line 157, in SynchronizedLock.synchronized
  File "synchronized_impl.dart", line 147, in SynchronizedLock.synchronized.run
  File "synchronized_impl.dart", line 83, in SynchronizedLock._run
  File "future.dart", line 222, in new Future.sync
  File "synchronized_impl.dart", line 84, in SynchronizedLock._run.<fn>
  File "zone.dart", line 1501, in runZoned
  File "zone.dart", line 1023, in _CustomZone.run
  File "zone.dart", line 1126, in _rootRun
  File "synchronized_impl.dart", line 86, in SynchronizedLock._run.<fn>.<fn>
  File "<asynchronous suspension>"
  File "flutter_cache_manager.dart", line 137, in CacheManager._cleanCache.<fn>
  File "<asynchronous suspension>"
  File "flutter_cache_manager.dart", line 153, in CacheManager._removeOldObjectsFromCache
  File "iterable.dart", line 438, in WhereIterator.moveNext
  File "compact_hash.dart", line 436, in _CompactIterator.moveNext
```

**Comment by renefloor** [comment]
I'll look into that

**Comment by renefloor** [comment]
This should be fixed in version `0.1.0-rc.1`.
